nix/overlay.nix

```diff
@@ -57,8 +57,8 @@ in {
   yarn = super.yarn.override { nodejs = super.nodejs-18_x; };
   openjdk = super.openjdk11_headless;
   xcodeWrapper = super.xcodeenv.composeXcodeWrapper {
-    version = "15.0";
-    allowHigher = true;
+    version = "15.1";
+    allowHigher = false;
```

Comment:
Why are we switching this to `false`? Do we intend to be more strict with newer Xcode versions from now on?

My experience is that older once cause issues, newer ones less so. That's why I added this flag.

Reply:
Yeah in our case both Xcode 14.3 and 15.3 cause build issues.
The only working one is 15.1 hence I made this more strict.

Reply:
Maybe a solution would be to be more exact and turn `version` into `versions` and provide a list?

Reply:
yes it was indeed better to use a `versions` list